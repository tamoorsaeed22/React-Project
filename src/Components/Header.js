import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../images/planet.png';
import './Header.module.css';

export default function Header() {
  const navStyles = ({ isActive }) => ({
    fontWeight: isActive ? 'bold' : 'normal',
    textDecoration: isActive ? 'underline' : 'none',
  });

  return (
    <header>
      <div className="header-logo-and-name">
        <img className="header-logo" src={logo} alt="planet logo" />
        <h1>Space Travelers&apos; Hub</h1>
      </div>
      <nav>
        <NavLink style={navStyles} to="/">Rockets</NavLink>
        <NavLink style={navStyles} to="/Missions">Missions</NavLink>
        <NavLink className="profile" style={navStyles} to="/MyProfile">My Profile</NavLink>
      </nav>
    </header>
  );
}
