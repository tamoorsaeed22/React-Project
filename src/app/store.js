import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './features/rocketsReducer';
import missionsReducer from './features/missionReducer';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
