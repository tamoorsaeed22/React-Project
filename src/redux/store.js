import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rocket/rocketsReducer';
import missionsReducer from './mission/missionReducer';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
