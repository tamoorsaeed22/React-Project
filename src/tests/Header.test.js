import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Header from '../Components/Header';

describe('Header test', () => {
  it('renders the Header', () => {
    const headerTest = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(headerTest).toMatchSnapshot();
  });
});
