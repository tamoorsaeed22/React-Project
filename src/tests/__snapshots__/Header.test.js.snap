exports[`Header test renders the Header 1`] = `
<header>
  <div
    className="header-logo-and-name"
  >
    <img
      alt="planet logo"
      className="header-logo"
      src="planet.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <nav>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "fontWeight": "bold",
          "textDecoration": "underline",
        }
      }
    >
      Rockets
    </a>
    <a
      className=""
      href="/Missions"
      onClick={[Function]}
      style={
        Object {
          "fontWeight": "normal",
          "textDecoration": "none",
        }
      }
    >
      Missions
    </a>
    <a
      className="profile"
      href="/MyProfile"
      onClick={[Function]}
      style={
        Object {
          "fontWeight": "normal",
          "textDecoration": "none",
        }
      }
    >
      My Profile
    </a>
  </nav>
</header>
`;